pkg/remote/remotes/docker/fetcher.go

```diff
@@ -209,7 +215,22 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest) (i
 	if getReq == nil {
 		// Fall back to the "manifests" endpoint
 		for _, host := range r.hosts {
-			getReq, sz, err = r.createGetReq(ctx, host, "manifests", dgst.String())
+			getReq, sz, err = r.createGetReq(ctx, host, mediatype, "manifests", dgst.String())
+			if err == nil {
+				break
+			}
+			// Store the error for referencing later
+			if firstErr == nil {
+				firstErr = err
+			}
+		}
+	}
+
+	if getReq == nil {
+		// Fall back to the "manifests" endpoint with specific mediatype
+		mediatype = "application/vnd.oci.image.manifest.v1+json"
+		for _, host := range r.hosts {
+			getReq, sz, err = r.createGetReq(ctx, host, mediatype, "manifests", dgst.String())
```

Comment:
Can we make a slice like `mediaTypes := []string{ images.MediaTypeDockerSchema2Manifest, images.MediaTypeDockerSchema2ManifestList, images.MediaTypeDockerSchema1Manifest, ocispec.MediaTypeImageManifest, ocispec.MediaTypeImageIndex }` ? And then add them into the `Accept` request header.

Reply:
Fixed